src/main/java/seedu/address/logic/commands/FilterPropertyCommand.java

```diff
@@ -0,0 +1,82 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LTE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TYPE;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.property.MatchingPrice;
+import seedu.address.model.property.Type;
+
+/**
+ * Filters properties based on their optional {@code PropertyType}.
+ */
+public class FilterPropertyCommand extends Command {
+    /** The command word to trigger the filtering action. */
+    public static final String COMMAND_WORD = "filterproperty";
+
+    /**
+     * Usage information for the filterclient command.
+     * Provides a description of the command's purpose and the format for entering client names.
+     */
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Filters the clients based on PropertyType and price. "
+            + "Parameters: PropertyType (must be a string) and/or price (must be an integer) "
+            + PREFIX_TYPE + "[TYPE]\n" + PREFIX_LTE + "[MATCHING_PRICE]\n"
+            + "Example: " + COMMAND_WORD + " " + PREFIX_TYPE + "CONDO" + " " + PREFIX_LTE + "30000000";
+
+    public static final String MESSAGE_SUCCESS = "Listed filtered properties";
+    public static final String MESSAGE_UNSUCCESS = "Filtered properties failed";
+
+    /** The type used for filtering. */
+    private Type type;
+    private MatchingPrice lteObj;
+    private MatchingPrice gteObj;
+
+    /**
+     * Constructs a FilterClientCommand to filter the specified {@code Property}.
+     *
+     * @param type The client to filter by.
+     * @throws NullPointerException If the provided client is null.
+     */
+    public FilterPropertyCommand(Type type, MatchingPrice lteObj, MatchingPrice gteObj) {
+        this.type = type;
+        this.lteObj = lteObj;
```

Comment:
maybe a comment on what lte/ gteObj means semantically will help with the documentation